pkg/app/piped/executor/waitapproval/waitapproval.go

```diff
@@ -161,3 +175,35 @@ func (e *Executor) getMentionedAccounts(event model.NotificationEventType) ([]st
 
 	return notification.FindSlackAccounts(event), nil
 }
+
+func (e *Executor) validateApproverNum(approver string) (string, bool) {
+	n, ok := e.MetadataStore.Stage(e.Stage.Id).Get(minApproverNum)
+	if !ok {
+		return "", false
+	}
+	num, err := strconv.Atoi(n)
+	if err != nil {
+		e.LogPersister.Errorf("%s could not be converted to integer: %v", num, err)
+		return "", false
+	}
+	if num <= 1 {
+		return approver, true
+	}
+	as, ok := e.MetadataStore.Stage(e.Stage.Id).Get(approversKey)
+	if !ok {
+		e.LogPersister.Infof("%d more approvals are needed", num-1)
+		return approver, false
+	}
+	approvers := strings.Split(as, ",")
+	for _, a := range approvers {
+		if a == approver {
+			e.LogPersister.Errorf("%s has already approved. %d more approvals are needed", num-1)
+			return "", false
+		}
+	}
+	if d := num - len(approvers) - 1; d > 0 {
+		e.LogPersister.Infof("%d more approvals are needed", d)
+		return as + "," + approver, false
+	}
+	return as + "," + approver, true
+}
```

Comment:
I think the log should be something like this to be more understandable, how do you think?

```
Waiting for approval from at least 2 users...

Got approval from "user-1"
Waiting for other approvers...

Got approval from "user-2"
This stage has been approved by 2 users ("user-1", "user-2")
```